test/cases/api.sh

```diff
@@ -326,6 +352,23 @@ case $(set +x; . /etc/os-release; echo "$ID-$VERSION_ID") in
     ;;
 esac
 
+# Only RHEL need subscription block.
+if [[ "$ID" == "rhel" ]]; then
```

Comment:
add ` && [ ! "$pretty_name" =~ 'Beta' ]`

then for Beta images SUBSCRIPTION_BLOCK is empty.

Reply:
About this part I'm also a little confused. It seems that the base machine and the guest VM are not the same RHEL versoin. See https://gitlab.com/osbuild/ci/osbuild-composer/-/jobs/1376400878 , at the beginning, the /etc/os-release shows the system is RHEL-8.3(https://gitlab.com/osbuild/ci/osbuild-composer/-/jobs/1376400878#L392):
```
$ schutzbot/deploy.sh
+ PROJECT=osbuild-composer
+ source /etc/os-release
++ NAME='Red Hat Enterprise Linux'
++ VERSION='8.3 (Ootpa)'
++ ID=rhel
```
But when I get the pretty_name from the guest VM it shows the system is RHEL-8.4 Beta:(https://gitlab.com/osbuild/ci/osbuild-composer/-/jobs/1376400878#L3390)
```
++ ssh -oStrictHostKeyChecking=no -i /tmp/tmp.f6VSJ5cul0/id_azure cloud-user@104.40.137.221 grep '^PRETTY_NAME=' /etc/os-release
+ pretty_name='PRETTY_NAME="Red Hat Enterprise Linux 8.4 Beta (Ootpa)"'
```

So that I cannot get the pretty_name before the guest VM is actually boot up.

Reply:
I'm not quite sure what happens here either but I'm sure it has to do with the fact that the content for the images is decoupled from the host operating system. It's probably coming from the repository json files shipped together with the RPM. 

But that goes to show another thing - you can't rely on the host OS to make these decisions. The decision should be based on some external variable which we're going to pass down to this test script in the test matrix.  Similar to how we pass INTERNAL_NETWORK.

This is becoming a mess quickly. 

Option 1) Split out the Azure boot up part of the code in a separate PR - that should not have any issues anymore and we can merge it as soon as CI reports pass. Leave the subscription portion here (after a rebase).

Option 2) Wait until we discuss and make a decision whether or not there's value to persue subscription testing on unreleased targets and then make a decision how to implement it to avoid all of these problems.

Reply:
Hi @atodorov ,
Thanks for your suggestion! I think use external variable is a good idea. But I don't wish to split this PR because there are already too many discussions/changes in this PR and I don't want to make it too messy. IMO the current solution(get the OS info from the guest) is not "wrong" but only not graceful enough. Shell we consider to accept the current solution and open a new issue to change it to the external variable way later? Thanks again!

Reply:
Few comments here: 

* I am proposing the split only of the boot portion of this test which doesn't appear to have issues and the majority of the discussion is not about that. So there's not much to lose in practicality here.  This is if you don't want the second part of this PR to hold-back a portion of this test script which appears to be functioning fine.

* I don't see the point of merging a sub-optimal implementation and then following with more changes when:  1) we've identified a clear problem and possible solution 2) the current PR is still open and 3) we are not in a hurry to merge the current PR immediately. We have spent a significant time here already and it doesn't really make any difference if we merge now or spend some more time to adjust the implementation given all of the new information discovered in the last few days.  1 or 2 PRs doesn't matter here, in any case the test script will be WIP.

Reply:
@jrusz, @ondrejbudai what are your preferences here ^^^ ?

Reply:
Thank you for your explanation @atodorov ! I agree that we can discuss more about the new external variable in the meeting. I've opened an issue(#1508)